src/capymoa/anomaly/autoencoder.py

```diff
@@ -0,0 +1,92 @@
+
+from capymoa.base import AnomalyDetector
+from capymoa.instance import Instance
+from capymoa.type_alias import AnomalyScore, LabelIndex
+import torch
+import torch.nn as nn
+import torch.optim as optim
+
+
+class Autoencoder(AnomalyDetector):
+    """ Autoencoder anomaly detector
+
+        This is a simple autoencoder anomaly detector that uses a single hidden layer.
+        The autoencoder is a duplicated version of MOA's Autoencoder class, but written in PyTorch."""
```

Comment:
Hi @justinuliu , is it sensible to include the following reference for this autoencoder implementation? The "duplicated version of MOA's autoencoder class" comment can be removed (or adapted to "This implementation is aligned with the one in MOA, but written using PyTorch")

Dong, Y., & Japkowicz, N. (2018). Threaded ensembles of autoencoders for stream learning. Computational Intelligence, 34(1), 261-281.